wdl-lint/src/rules/call_input_spacing.rs

```diff
@@ -0,0 +1,174 @@
+        .with_fix("add newline before the input")
+}
+
+/// Creates call input assignment diagnostic.
+fn call_input_assignment(span: Span) -> Diagnostic {
+    Diagnostic::warning("call inputs assignments must be surrounded with whitespace")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("surround '=' with whitespace on each side")
+}
+
+/// Detects unsorted input declarations.
+#[derive(Debug, Clone, Copy)]
+pub struct CallInputSpacingRule;
+
+impl Rule for CallInputSpacingRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures that call inputs are spaced appropriately."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "When making calls from a workflow, it is more readable and easier to edit if the supplied \
+         inputs are each on their own line. When there is more than one input to a call statement, \
+         the `input:` keyword should follow the opening brace ({) and a single space, then each \
+         input specification should occupy its own line. This does inflate the line count of a WDL \
+         document, but it is worth it for the consistent readability. An exception can be made \
+         (but does not have to be made), for calls with only a single parameter. In those cases, \
+         it is permissable to keep the input on the same line as the call."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Style, Tag::Clarity, Tag::Spacing])
+    }
+}
+
+impl Visitor for CallInputSpacingRule {
+    type State = Diagnostics;
+
+    fn call_statement(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        call: &CallStatement,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        let inputs = call.inputs().count();
+
+        if inputs == 0 {
+            return;
+        }
+
+        // Check for "{ input:" spacing
+        let mut spacing_errors = 0;
+        let input_keyword = call
+            .syntax()
+            .children_with_tokens()
+            .find(|c| c.kind() == SyntaxKind::InputKeyword);
+
+        if let Some(rbrace) = call
+            .syntax()
+            .children_with_tokens()
+            .find(|c| c.kind() == SyntaxKind::OpenBrace)
+        {
+            match input_keyword.borrow() {
+                Some(t) => {
+                    if let Some(next) = rbrace.next_sibling_or_token() {
+                        if next.as_token().expect("should be a token").text().eq(" ") {
+                            if let Some(second) = next.next_sibling_or_token() {
+                                if second != *t {
+                                    spacing_errors += 1;
+                                }
+                            }
+                        } else {
+                            spacing_errors += 1;
+                        }
+                    }
+                }
```

Comment:
I don't understand this block

Reply:
I'm finding the `OpenBrace`, then checking to ensure the next token is exactly one space (" "). Then looking to the next node/token and ensuring it matches the `InputKeyword` that was previously found.

Reply:
Could we not instead just look for the `input` keyword, get its previous sibling (with tokens) and:

* if the previous sibling is whitespace, ensure it is `" "`; otherwise if not `" "`, emit a diagnostic pointing at the span of the whitespace saying "change this whitespace to a single space".
* if the previous sibling is not whitespace, emit a diagnostic pointing at the input keyword saying "add a space before this" (or alternatively, label the span of the previous sibling and say "add a space after this").

Reply:
Ah, I suppose anything other than whitespace would be a validation error before linting. So we could just check that the prior token is whitespace and is `" "`.